bokehjs/src/lib/models/glyphs/segment.ts

```diff
@@ -116,6 +116,20 @@ export class SegmentView extends GlyphView {
     for (const i of candidates) {
       if ((v0[i] <= val && val <= v1[i]) || (v1[i] <= val && val <= v0[i]))
         hits.push(i)
+
+      const threshold = 2.5 // Maximum pixel difference to detect hit
```

Comment:
We should probably try computing this value as the `line_width/2` plus some padding(maybe ~1.5?) that way this should work with segments that are "thick" (i.e. more than 1 pixel wide)